core/src/main/java/org/apache/calcite/tools/Frameworks.java

```diff
@@ -143,19 +155,33 @@ public R apply(RelOptCluster cluster, RelOptSchema relOptSchema,
    * @param action Callback containing user-specified code
    * @return Return value from action
    */
-  public static <R> R withPrepare(PrepareAction<R> action) {
+  public static <R> R withPrepare(BasePrepareAction<R> action) {
```

Comment:
I'm ok with this change and just notice that `action` will run with the "default" `FrameworkConfig` with this method.